### PR TITLE
[PL-46673] Hide log info using regex - feature and delegate HF doc

### PR DESCRIPTION
**Pull request by brian-f-harness**
Hide log info using regex

For reviewers, previews are available:

[Hide log information based on regex patterns](https://65b7db0bdd6169228b1d5b81--harness-developer.netlify.app/docs/platform/delegates/manage-delegates/hide-logs-using-regex)

https://65b7db0bdd6169228b1d5b81--harness-developer.netlify.app/docs/faqs/harness-delegate-faqs#can-i-remove-sensitive-information-from-logs

HF release note: https://65b7db0bdd6169228b1d5b81--harness-developer.netlify.app/release-notes/delegate#version-240182110

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/platform/delegates/manage-delegates/hide-logs-using-regex.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+title: Hide log information based on regex patterns
+description: This topic describes how to sensitive log information based on regex patterns.
+sidebar_position: 9
+---
+
+You can mask sensitive information from log streams using regular expressions. This will remove the information you define from logs in the Harness UI.
+
+To hide log information based on regex, do the following:
+
+1. Create a `sanitize-patterns.txt` file in your local that contains all of the regex that you want to hide in logs.
```
Comment (brian-f-harness) [inline docs/platform/delegates/manage-delegates/hide-logs-using-regex.md:11]:
```suggestion
1. Create a `sanitize-patterns.txt` file in your local that contains all of the regexes that you want to hide in logs.
```

File: docs/platform/delegates/manage-delegates/hide-logs-using-regex.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+   This file can contain multiple expressions. Add each regex on a new line. In the example below, we have added two expressions.
```
Comment (brian-f-harness) [inline docs/platform/delegates/manage-delegates/hide-logs-using-regex.md:13]:
```suggestion
   This file can contain multiple regexes. Add each regex on a new line. In the example below, we have added two expressions.
```

File: docs/platform/delegates/manage-delegates/hide-logs-using-regex.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+   <accountID>.*?</accountID>
+   ```
+
+2. Create the ConfigMap in the same namespace where you are installing the delegate. 
+
+   ```
+   kubectl create configmap <CONFIGMAP_NAME> --from-file sanitize-patterns.txt -n <NAMESPACE>         
+   ```
+
+3. Mount the volume under the `/opt/harness-delegate` path.
```
Comment (brian-f-harness) [inline docs/platform/delegates/manage-delegates/hide-logs-using-regex.md:26]:
```suggestion
3. Mount the volume under the `/opt/harness-delegate` path in your delegate YAML file.
```

File: docs/platform/delegates/manage-delegates/hide-logs-using-regex.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+3. Mount the volume under the `/opt/harness-delegate` path in your delegate YAML file.
+
+   ```yaml
+               volumeMounts:
+              - name: "config"
+                mountPath: "/opt/harness"
+          volumes:
+            - name: "config"
+              configMap:
+                name: "config-sanitize"
```
Comment (bharatgoelharness) [inline docs/platform/delegates/manage-delegates/hide-logs-using-regex.md:36]: name: <CONFIGMAP_NAME>

**Review by bharatgoelharness — DISMISSED** [review]

File: docs/platform/delegates/manage-delegates/hide-logs-using-regex.md (hunk 4 above)
Comment (brian-f-harness) [inline docs/platform/delegates/manage-delegates/hide-logs-using-regex.md:36]:
```suggestion
                name: "<YOUR_CONFIGMAP_FILE_NAME>"
```

**Review by bharatgoelharness — APPROVED** [review]

**Pull request merged**